// nsfw-ts/dist/NSFW.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NSFW = void 0;
var superagent = require("superagent");
var NSFW = /** @class */ (function () {
    function NSFW() {
        this.version = require("../package.json").version;
        this.methods = [
            "anal",
            "fourk",
            "ass",
            "gonewild",
            "pgif",
            "pussy",
            "thigh",
            "boobs",
            "blowjob",
            "hentaiass",
            "hentai",
            "hmidriff",
            "hentaithigh",
            "erokemo",
            "kitsune",
            "lewd",
            "nekofeet",
            "nekopussy",
            "nekotits",
            "solo",
            "wallpaper",
        ];
    }
    NSFW.prototype.anal = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=anal")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.fourk = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=4k")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.ass = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=ass")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.gonewild = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=gonewild")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.pgif = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=pgif")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.pussy = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=pussy")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.thigh = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=thigh")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.boobs = function () {
        return __awaiter(this, void 0, void 0, function () {
            var id, res, preview, image;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        id = [Math.floor(Math.random() * 10930)];
                        return [4 /*yield*/, superagent.get("http://api.oboobs.ru/boobs/".concat(id))];
                    case 1:
                        res = _a.sent();
                        preview = res.body[0]["PREVIEW".toLowerCase()];
                        image = "http://media.oboobs.ru/".concat(preview);
                        return [2 /*return*/, image];
                }
            });
        });
    };
    NSFW.prototype.hentaiass = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=hass")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.hentai = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=hentai")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.hmidriff = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=hmidriff")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.hentaithigh = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekobot.xyz/api/image?type=hthigh")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.message];
                }
            });
        });
    };
    NSFW.prototype.erokemo = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekos.life/api/v2/img/erokemo")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.url];
                }
            });
        });
    };
    NSFW.prototype.kitsune = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekos.life/api/v2/img/fox_girl")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.url];
                }
            });
        });
    };
    NSFW.prototype.lewd = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekos.life/api/v2/img/lewd")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.url];
                }
            });
        });
    };
    NSFW.prototype.nekofeet = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekos.life/api/v2/img/feet")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.url];
                }
            });
        });
    };
    NSFW.prototype.nekopussy = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekos.life/api/v2/img/pussy_jpg")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.url];
                }
            });
        });
    };
    NSFW.prototype.nekotits = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekos.life/api/v2/img/tits")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.url];
                }
            });
        });
    };
    NSFW.prototype.solo = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekos.life/api/v2/img/solo")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.url];
                }
            });
        });
    };
    NSFW.prototype.wallpaper = function () {
        return __awaiter(this, void 0, void 0, function () {
            var body;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, superagent.get("https://nekos.life/api/v2/img/wallpaper")];
                    case 1:
                        body = (_a.sent()).body;
                        return [2 /*return*/, body.url];
                }
            });
        });
    };
    return NSFW;
}());
exports.NSFW = NSFW;
